e2e-tests/IIoTPlatform-E2E-Tests/TestHelper.cs

```diff
@@ -132,23 +148,33 @@ internal static class TestHelper {
         /// <summary>
         /// Switch to publisher standalone mode
         /// </summary>
-        public static void SwitchToStandaloneMode(IIoTPlatformTestContext context) {
+        /// <param name="context">Shared Context for E2E testing Industrial IoT Platform</param>
+        /// <param name="ct">Cancellation token</param>
+        public static async Task SwitchToStandaloneModeAsync(
+            IIoTPlatformTestContext context,
+            CancellationToken ct = default
+        ) {
             var patch =
                 @"{
                     tags: {
                         unmanaged: true
                     }
                 }";
 
-            UpdateTagAsync(patch, context).Wait();
+            await UpdateTagAsync(patch, context, ct);
         }
 
         /// <summary>
         /// Switch to publisher orchestrated mode
         /// </summary>
         /// <param name="destinationFilePath">Path of the PublishedNodesFile.json file to be deleted</param>
         /// <param name="context">Shared Context for E2E testing Industrial IoT Platform</param>
-        public static void SwitchToOrchestratedMode(string destinationFilePath, IIoTPlatformTestContext context) {
+        /// <param name="ct">Cancellation token</param>
+        public static async Task SwitchToOrchestratedModeAsync(
+            string destinationFilePath,
+            IIoTPlatformTestContext context,
+            CancellationToken ct = default
```

Comment:
I would remove all the `= default` if we are going to support CancellationToken than the caller should be forced to decide to use it (or not), meaning the caller has to pass default or a concrete instance. This comment is valid for almost every modified method.

Reply:
I don't agree with that. I will argue that providing the default is the right way to give a choice of using `CancellationToken` or not. If the user of method choses not to use a `CancellationToken`, then they should not be forced into writing additional code if there is a possibility of not doing that.